shhext/service_test.go

```diff
@@ -0,0 +1,144 @@
+package shhext
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestShhExtSuite(t *testing.T) {
+	suite.Run(t, new(ShhExtSuite))
+}
+
+type ShhExtSuite struct {
+	suite.Suite
+
+	nodes    []*node.Node
+	services []*Service
+	whisper  []*whisper.Whisper
+}
+
+func (s *ShhExtSuite) SetupTest() {
+	s.nodes = make([]*node.Node, 2)
+	s.services = make([]*Service, 2)
```

Comment:
It seems like only one `Service` and one `Whisper` is used. It's fine but maybe something like this would better explain the intention:
```go
s.node := node.New(&node.Config{...})
s.whisper := whisper.New(nil)
s.service := New(...)

node.Register(...)
node.Register(...)

peer := node.New(&node.Config{...})
node.Server().AddPeer(peer)
```

Reply:
whisper needs to be enabled for every peer , otherwise they won't connect

Reply:
Yeah, it needs to be added to `peer` as well, I missed that. But it's not used in the test cases.